Invoke-Locksmith.ps1

```diff
@@ -87,6 +87,54 @@ $PreferredOwner = New-Object System.Security.Principal.SecurityIdentifier($Enter
 # $AdminUsers = $Admins | ForEach-Object { (Get-ADGroupMember $_ | Where-Object { $_.objectClass -eq 'user'}).SamAccountName } | Select-Object -Unique
 # $AdminUsers | ForEach-Object { $SafeUsers += "|$($env:USERDOMAIN)\\" + $_ }
 
+function Test-IsElevated {
+    <#
+    .SYNOPSIS
+        Tests if PowerShell is running with elevated privileges (run as Administrator).
+    .DESCRIPTION
+        This function returns True if the script is being run as an administrator or False if not.
+    .EXAMPLE
+        Test-IsElevated
+    .EXAMPLE
+        if (!(Test-IsElevated)) { Write-Host "You are not running with elevated privileges and will not be able to make any changes." -ForeGroundColor Yellow }
+    #>
+    $identity = [Security.Principal.WindowsIdentity]::GetCurrent()
+    $principal = New-Object Security.Principal.WindowsPrincipal $identity
+    $principal.IsInRole([Security.Principal.WindowsBuiltinRole]::Administrator)
+
+    <# Optional: Prompt to launch elevated if not already running as administrator:
+    if (-not ( [Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator") ) {
+        $arguments = "& '" + $myinvocation.mycommand.definition + "'" Start-Process powershell -Verb runAs -ArgumentList $arguments Break
+    }
+    #>
+}
+
+
+function Test-IsDomainAdmin {
+    <#
+    .SYNOPSIS
+        Tests if the current user has Domain Admin rights (or a comparable role).
+    .DESCRIPTION
+        This function returns True if the current user is a Domain Admin or False if not.
+    .EXAMPLE
+        Test-IsDomainAdmin
+    .EXAMPLE
+        if (!(Test-IsDomainAdmin)) { Write-Host "You are not running with Domain Admin rights and will not be able to make certain changes." -ForeGroundColor Yellow }
+    #>
+    if (-not (
```

Comment:
Let's re-write this function to remove the -not  from the check.

Reply:
To clarify:
We should check if the user **is** a member of those groups instead of checking if the user is not a member of the groups.